pytools/py_codegen.py

```diff
@@ -35,42 +41,68 @@
 
 
 class PythonCodeGenerator(CodeGeneratorBase):
-    def get_module(self, name=None):
+    def get_module(self, name: str | None = None,
+                   _from_get_function: bool = False) -> dict[str, Any]:
         if name is None:
             name = "<generated code>"
 
-        result_dict = {}
+        result_dict: dict[str, Any] = {}
         source_text = self.get()
+
+        # {{{ Handle Python's linecache
+
+        import linecache
+
+        if name in linecache.cache:
+            from warnings import warn
+            warn(f"Overwriting existing generated code in linecache: '{name}'.",
+                    stacklevel=3 if _from_get_function else 2)
+
+        linecache.cache[name] = (None, None,  # type: ignore[assignment]  # pyright: ignore[reportArgumentType]
+                                 [e+"\n" for e in source_text.split("\n")], None)
+
+        # }}}
+
         exec(compile(
             source_text.rstrip()+"\n", name, "exec"),
                 result_dict)
+
+        # For pudb:
         result_dict["_MODULE_SOURCE_CODE"] = source_text
```

Comment:
I think pudb is *already* perfectly happy with `linecache`, so this can probably just go. (Please try.)

Reply:
Yep, that seems to work! Removed in [a11b3f8](https://github.com/inducer/pytools/pull/295/commits/a11b3f84b81c280d77a7ddbd0d759121551594dd)